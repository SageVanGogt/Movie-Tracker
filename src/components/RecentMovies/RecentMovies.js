import React from 'react';
import {connect} from "react-redux"
import Card from './../Card/Card'

const RecentMovies = (props) => {
  const {user, films} = props
  const allMovies = films.map(film => {
    return (
      <Card {...film}/>
    )
  })
  return (
    <div>
      {allMovies}
    </div>
  )
}

export const mapStateToProps = (state) => ({
  user: state.user,
  films: state.films
})

export default connect(mapStateToProps, null)(RecentMovies);