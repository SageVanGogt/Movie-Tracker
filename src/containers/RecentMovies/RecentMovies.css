.movies-container {
  padding-top: 120px;
  display: flex;
  flex-wrap: wrap;
  align-content: space-between;
  justify-content: center;
}