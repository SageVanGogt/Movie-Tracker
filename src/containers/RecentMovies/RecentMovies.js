import React from 'react';
import { connect } from "react-redux";
import Card from './../../components/Card/Card';
import PropTypes from 'prop-types';
import './RecentMovies.css'

const RecentMovies = (props) => {
  const {user, films} = props
  const allMovies = films.map(film => {
    return (
      <Card {...film} user={user} key={film.movie_id}/>
    )
  })
  return (
    < div className="movies-container" >
      {allMovies}
    </div>
  )
}

export const mapStateToProps = (state) => ({
  user: state.user,
  films: state.films
})

RecentMovies.propTypes = {
  films: PropTypes.array,
  user: PropTypes.object
}

export default connect(mapStateToProps, null)(RecentMovies);