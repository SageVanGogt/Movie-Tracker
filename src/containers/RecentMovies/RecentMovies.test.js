import React from 'react';
import {RecentMovies, mapStateToProps} from './RecentMovies';
import {shallow} from 'enzyme';

describe('RecentMovies', () => {
  it('should match the snapshot', ()=> {
    let mockProps = {
      films:[{movie_id:1}, {movie_id:2}, {movie_id:3}],
      user:{
        user_id:1
      }
    };
    let wrapper = shallow(<RecentMovies {...mockProps}/>);

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should map the user to props', () => {
      const mockState = {
        user: {
          id: 1,
          name: 'Alan'
        },
        films: [{}, {}, {}]
      };

      const expected = {
        user: {
          id: 1,
          name: 'Alan'
        },
        films: [{}, {}, {}]
      };
      
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

});