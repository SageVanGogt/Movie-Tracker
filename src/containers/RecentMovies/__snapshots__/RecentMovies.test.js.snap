// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentMovies should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    films={
        Array [
            Object {
              "movie_id": 1,
            },
            Object {
              "movie_id": 2,
            },
            Object {
              "movie_id": 3,
            },
          ]
    }
    user={
        Object {
            "user_id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component)
          movie_id={1}
          user={
                    Object {
                              "user_id": 1,
                            }
          }
/>,
        <Connect(Component)
          movie_id={2}
          user={
                    Object {
                              "user_id": 1,
                            }
          }
/>,
        <Connect(Component)
          movie_id={3}
          user={
                    Object {
                              "user_id": 1,
                            }
          }
/>,
      ],
      "className": "movies-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "movie_id": 1,
          "user": Object {
            "user_id": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "movie_id": 2,
          "user": Object {
            "user_id": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "movie_id": 3,
          "user": Object {
            "user_id": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component)
            movie_id={1}
            user={
                        Object {
                                    "user_id": 1,
                                  }
            }
/>,
          <Connect(Component)
            movie_id={2}
            user={
                        Object {
                                    "user_id": 1,
                                  }
            }
/>,
          <Connect(Component)
            movie_id={3}
            user={
                        Object {
                                    "user_id": 1,
                                  }
            }
/>,
        ],
        "className": "movies-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "movie_id": 1,
            "user": Object {
              "user_id": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "movie_id": 2,
            "user": Object {
              "user_id": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "movie_id": 3,
            "user": Object {
              "user_id": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
